File: #0452.minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {

        if (points.size() == 0) {
            return 0;
        }

        std::sort(points.begin(), points.end(), 
            [](vector<int>& a, vector<int>& b) {
                return a[1] < b[1];
        });

        int index = points[0][1];
        int result = 1;
        for (vector<int>& point : points) {
            if (point[0] <= index) {
                continue;
            }
            result++;
            index = point[1];
        }
        return result;
    }
};

File: #0452.minimum-number-of-arrows-to-burst-balloons.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
    let count = 0;
    let min = null;
    points.sort((left, right) => left[0] - right[0]);
    for (let [x, y] of points) {
        if (min == null || min < x) {
            count++
            min = y
        } else {
            min = Math.min(min, y);
        }
    }
    return count;
};
